pyEpiabm/pyEpiabm/tests/test_sweep/test_host_progression_sweep.py

```diff
@@ -13,34 +14,34 @@ def setUp(self) -> None:
         """Sets up a population we can use throughout the test.
         2 people are located in one microcell.
         """
-        self.test_population = pe.Population()
-        self.test_population.add_cells(1)
-        self.test_population.cells[0].add_microcells(1)
-        self.test_population.cells[0].microcells[0].add_people(3)
-        self.person1 = self.test_population.cells[0].microcells[0].persons[0]
-        self.person2 = self.test_population.cells[0].microcells[0].persons[1]
-        self.person3 = self.test_population.cells[0].microcells[0].persons[2]
+        # Create population that will be used to test all
+        #  methods except update status
+        self.test_population1 = pe.Population()
+        self.test_population1.add_cells(1)
+        self.test_population1.cells[0].add_microcells(1)
+        self.test_population1.cells[0].microcells[0].add_people(3)
+        self.person1 = self.test_population1.cells[0].microcells[0].persons[0]
+        self.person2 = self.test_population1.cells[0].microcells[0].persons[1]
+        self.person3 = self.test_population1.cells[0].microcells[0].persons[2]
+
+        # Create a population with people of all infection statuses to
+        # test update status method
+        self.test_population2 = pe.Population()
+        self.test_population2.add_cells(1)
+        self.test_population2.cells[0].add_microcells(1)
+        self.test_population2.cells[0].microcells[0].\
+            add_people(len(InfectionStatus))
+        self.people = []
+        for i in range(len(InfectionStatus)):
+            person = self.test_population2.cells[0].microcells[0].persons[i]
+            person.update_status(InfectionStatus(i + 1))
+            self.people.append(person)
 
     def test_construct(self):
         """Tests that the host progression sweep initialises correctly.
         """
         pe.sweep.HostProgressionSweep()
 
-    def test_update_status(self):
-        """Tests the update status function on the test population. Person 2
-        is set to infected and we check that their status changes to Recovered
-        on the next time step.
-        """
-        self.person2.update_status(pe.property.InfectionStatus.InfectMild)
-        test_sweep = pe.sweep.HostProgressionSweep()
-        test_sweep.bind_population(self.test_population)
-        test_sweep._update_next_infection_status(self.person2)
-        self.assertEqual(self.person2.next_infection_status,
-                         pe.property.InfectionStatus.Recovered)
-        self.person2.update_status(pe.property.InfectionStatus.Recovered)
-        self.assertRaises(TypeError, test_sweep._update_next_infection_status,
-                          self.person2)
-
     def test_set_latent_time(self):
```

Comment:
Use current time in L53 `assertTrue` for readability (or better still add current_time as an argument to the test (with a default value of 5.0 so you don't have to change anything else).